python/ray/dataframe/dataframe.py

```diff
@@ -1474,16 +1510,25 @@ def from_pandas(df, npartitions=None, chunksize=None, sort=True):
     elif chunksize is None:
         raise ValueError("The number of partitions or chunksize must be set.")
 
+    old_index = df.index
+
     # TODO stop reassigning df
     dataframes = []
+    lengths = []
     while len(df) > chunksize:
-        top = ray.put(df[:chunksize])
+        t_df = df[:chunksize]
+        lengths.append(len(t_df))
+        t_df.reindex()
```

Comment:
We are reindexing here because we want the dataframes to have a `pd.RangeIndex` within the partitions. It is smaller and sometimes faster.